browser/android/preferences/background_video_playback_tab_helper.cc

```diff
@@ -80,4 +84,74 @@ void BackgroundVideoPlaybackTabHelper::DidFinishNavigation(
   }
 }
 
+namespace chrome {
+namespace android {
+
+mojo::AssociatedRemote<script_injector::mojom::ScriptInjector> GetRemote(
+    content::RenderFrameHost* rfh) {
+  mojo::AssociatedRemote<script_injector::mojom::ScriptInjector>
+      script_injector_remote;
+  rfh->GetRemoteAssociatedInterfaces()->GetInterface(&script_injector_remote);
+  return script_injector_remote;
+}
+void JNI_BackgroundVideoPlaybackTabHelper_SendOrientationChangeEvent(
+    JNIEnv* env,
+    const base::android::JavaParamRef<jobject>& jweb_contents,
+    jboolean is_full_screen) {
+  content::WebContents* web_contents =
+      content::WebContents::FromJavaWebContents(jweb_contents);
+  // Injecting this script to make youtube video fullscreen on landscape mode
+  // and exit fullscreen on portrait mode.
+  std::string script;
+  if (is_full_screen) {
+    script =
+        "if(!document.fullscreenElement) {"
+        "   var fullscreenBtn = "
+        "     document.getElementsByClassName('fullscreen-icon');"
+        "   if(fullscreenBtn && fullscreenBtn.length > 0) {"
+        "      fullscreenBtn[0].click();"
+        "   } else {"
+        "     var moviePlayer = document.getElementById('movie_player');"
+        "     if (moviePlayer) {"
+        "         moviePlayer.click();"
+        "     }"
+        "     setTimeout(() => {"
```

Comment:
why does this use setTimeout?

Reply:
Youtube is not loading controllers initially for this usecase and because of that we are using setTimeout.